zaza/openstack/charm_tests/ceph/tests.py

```diff
@@ -690,28 +702,162 @@ def expected_apps(self):
     def multisite(self):
         """Determine whether deployment is multi-site."""
         try:
-            zaza_model.get_application('slave-ceph-radosgw')
+            zaza_model.get_application(self.secondary_rgw_app)
             return True
         except KeyError:
             return False
 
+    def get_rgwadmin_cmd_skeleton(self, unit_name:str):
+        '''Get radosgw-admin cmd skeleton with rgw.hostname populated key'''
+        app_name = unit_name.split('/')[0]
+        juju_units = zaza_model.get_units(app_name)
+        unit_hostnames = generic_utils.get_unit_hostnames(juju_units)
+        hostname = unit_hostnames[unit_name]
+        return 'radosgw-admin --id=rgw.{} '.format(hostname)
+
     @tenacity.retry(wait=tenacity.wait_exponential(multiplier=10, max=300),
                     reraise=True, stop=tenacity.stop_after_attempt(10),
                     retry=tenacity.retry_if_exception_type(AssertionError))
-    def wait_for_sync(self, application):
+    def wait_for_sync(self, application, isPrimary=False):
         """Wait for slave to secondary to show it is in sync."""
         juju_units = zaza_model.get_units(application)
         unit_hostnames = generic_utils.get_unit_hostnames(juju_units)
-        sync_states = []
-        sync_check_str = 'data is caught up with source'
+        sync_states = list()
+        data_check = 'data is caught up with source'
+        meta_primary = 'metadata sync no sync (zone is master)'
+        meta_secondary = 'metadata is caught up with master'
+        meta_check = meta_primary if isPrimary else meta_secondary
         for unit_name, hostname in unit_hostnames.items():
             key_name = "rgw.{}".format(hostname)
-            cmd = 'radosgw-admin --id={} zone sync status'.format(key_name)
+            cmd = 'radosgw-admin --id={} sync status'.format(key_name)
             stdout = zaza_model.run_on_unit(unit_name, cmd).get('Stdout', '')
-            sync_states.append(sync_check_str in stdout)
+            sync_states.append((data_check in stdout) and
+                               (meta_check in stdout))
         assert all(sync_states)
 
-    def test_processes(self):
+    @tenacity.retry(wait=tenacity.wait_exponential(multiplier=1, max=60),
+                    reraise=True, stop=tenacity.stop_after_attempt(12))
+    def fetch_rgw_object(self, target_client, container_name, object_name):
+        '''
+        Fetch RGW object content
+        :param target_client: boto3 client object configured for an endpoint.
```

Comment:
Can you include type information as well as descriptions about what the params are? I'm happy with either python typing or adding it via docstring formatted bits ([an example](https://github.com/openstack-charmers/zaza-openstack-tests/blob/master/zaza/openstack/utilities/ceph.py#L22-L23)), but we have, so far, kind of standardised on the docstrings

Reply:
Added docstring formatted info for all newly added methods.